Refactoring and final javadoc additions

Bensonwu02 opened the pull request:
This pull request contains
- final javadoc additions
- updated README
- case handled for non-numerical data input for goal setting and add metric
- addition of 'example metrics' folder for more example data.

Bensonwu02 committed:
Implemented goal tracking feature
- Used decorator design pattern to modify functionality of metric summary to display performance with respect to tracked goal

Adjusted UI components
- font of titles
- System exit when application is closed (in addition to frame dispose)

Adapted Import and Export use cases to fit goal tracking
- two new numbers stored in header of csv files
- has goal (true/false) is encoded as 1 or 0
- goal follows has goal in csv file
- import and export test cases pass and tested on running program. works for goal tracking now

Adapted Metric entity to fit goal tracking
- added new instance variables hasgoal and goal
- they have setters and getters for the use case
- added additional overloaded constructor for import and export since we will now need to keep track of hasgoal and goal in the csv files

Removed back button on Home Screen

Bensonwu02 committed:
Added setMetricGoal method to metric storage and metricstorageinterface
- so I do not need to access metric directly through my use case
- done through metric storage interface instead

Implemented message in metric summary screen to show dialog when no metrics are being tracked

Bensonwu02 committed:
amendment to test case.
- formatting issue was causing test to fail.

Bensonwu02 committed:
refactored metric delete test so it is in same package as other testing classes

Bensonwu02 committed:
Changed metric sum to allow viewing of graph as long as metric contains at least 1 datapoint

Bensonwu02 committed:
Changed test case so it fits changes from previous push.

Bensonwu02 committed:
Merge branch 'main' of https://github.com/CSC207-2022F-UofT/course-project-mewdy into FEATURE-Goal-tracking

 Conflicts:
	src/test/java/MetricDeleterTest.java

Bensonwu02 committed:
Made requested changes
- Back button now displays 'home' instead of 'back'
- Fixed controller so it does not access entities.
- Deleted MetricsumWithGoalController, now application only has one controller for metric summary containing decorated metric summarizer. Metric summarizer calls extended summarize metric only if metric contains a goal.
- Changed try catch block in choose metric sum - eliminated second try catch block
- added helper method to construct metric summary screen based on what kind of view model is given. (Construct with or without goal performance)

Bensonwu02 committed:
Made requested changes
- Back button now displays 'home' instead of 'back'
- Fixed controller so it does not access entities.
- Deleted MetricsumWithGoalController, now application only has one controller for metric summary containing decorated metric summarizer. Metric summarizer calls extended summarize metric only if metric contains a goal.
- Changed try catch block in choose metric sum - eliminated second try catch block
- added helper method to construct metric summary screen based on what kind of view model is given. (Construct with or without goal performance)

Bensonwu02 committed:
Made requested changes
- Back button now displays 'home' instead of 'back'
- Fixed controller so it does not access entities.
- Deleted MetricsumWithGoalController, now application only has one controller for metric summary containing decorated metric summarizer. Metric summarizer calls extended summarize metric only if metric contains a goal.
- Changed try catch block in choose metric sum - eliminated second try catch block
- added helper method to construct metric summary screen based on what kind of view model is given. (Construct with or without goal performance)

RE-mpire committed:
Merge branch 'main' of https://github.com/CSC207-2022F-UofT/course-project-mewdy into FEATURE-Goal-tracking

Bensonwu02 committed:
Merge branch 'main' of https://github.com/CSC207-2022F-UofT/course-project-mewdy into FEATURE-Goal-tracking

Bensonwu02 committed:
Merge branch 'main' of https://github.com/CSC207-2022F-UofT/course-project-mewdy into FEATURE-Goal-tracking

RE-mpire committed:
Updated error messages

Bensonwu02 committed:
Added Javadocs for set goal use case.
Updated README
Handled case where non-numerical data is given for add metric and set goal

Bensonwu02 committed:
Merge branch 'main' of https://github.com/CSC207-2022F-UofT/course-project-mewdy into FEATURE-Goal-tracking

Bensonwu02 committed:
Updated java docs
Updated README
Handled case where non-numerical data is given for add metric and set goal
Refactoring for final variables

Bensonwu02 committed:
Merge remote-tracking branch 'origin/FEATURE-Goal-tracking' into FEATURE-Goal-tracking

Bensonwu02 committed:
Updated java docs
Updated README
Handled case where non-numerical data is given for add metric and set goal
Refactoring for final variables
Example metric data added in new directory

Bensonwu02 committed:
Small update to the README to explain example metrics folder

craftedswift commented on src/main/java/models/SetGoalRequestModel.java:
```diff
@@ -1,16 +1,32 @@
+    /**
+     * Getter for metric name
+     * @return  metric name string
+     */
```
for java doc make sure there is a space between the method description and the @param/@return

Bensonwu02 replied:
resolved

craftedswift commented on src/main/java/presenters/MetricSumViewModel.java:
```diff
@@ -41,14 +41,30 @@ public XYChart getChart() {
     public String getMetricStatistics() {
         return metricStatistics;
     }
+
```
for java doc make sure there is a space between the method description and the @param/@return

craftedswift commented on src/main/java/use_cases/GoalSetter.java:
```diff
@@ -15,6 +17,11 @@ public GoalSetter(MetricStorageInterface metricStorage, SetGoalOutputBoundary pr
     }
 
+    /**
+     * Method for setting goal.
```
for java doc make sure there is a space between the method description and the @param/@return

Bensonwu02 replied:
resolved

tylerkj1209 reviewed (dismissed):
Thank you for altering the fail view message for dataLogger to more clearly match with the actual exception being raised by the code. I see no issues with this pull request. Lots of good reworking of the documentation.

craftedswift commented on src/main/java/screens/ChooseMetricSumScreen.java:
```diff
@@ -74,6 +74,12 @@ public void actionPerformed(ActionEvent evt) {
 
+    /**
+     * A helper method to extract data contained in the view model which is used to make the MetricSummaryScreen.
+     * @param viewModel represents view model from which this method extracts information to construct
```
for java doc make sure there is a space between the method description and the @param/@return

Bensonwu02 replied:
resolved

craftedswift commented on src/main/java/use_cases/MetricDelInputBoundary.java:
```diff
@@ -3,6 +3,15 @@
 public interface MetricDelInputBoundary {
+
+    /**
+     *
```
Missing method description (my bad)

Bensonwu02 replied:
resolved

Bensonwu02 commented on src/main/java/use_cases/SetGoalInputBoundary.java:
```diff
@@ -3,8 +3,16 @@
 public interface SetGoalInputBoundary {
 
+    /**
+     *
```
resolved

Bensonwu02 committed:
Small update to the README to explain example metrics folder

Bensonwu02 commented:
Made changes in accordance with sanford's comments

Wizard603 commented on README.md:
```diff
@@ -1,45 +1,39 @@
+Mewdy is a configurable metric tracking application.
 
 ## How to run:
 - run Main.java
```
Maybe mention there is a downloadable jar file in the releases tab

Bensonwu02 replied:
done

Wizard603 commented on src/main/java/controllers/DataExportController.java:
```diff
@@ -8,15 +8,15 @@
  */
 public class DataExportController {
-    private final DataExportInputBoundary INPUTBOUNDARY;
+    private final DataExportInputBoundary inputBoundary;
```
this is final so it should be all caps

Bensonwu02 replied:
only static variables are caps

Wizard603 commented on src/main/java/controllers/DataImportController.java:
```diff
@@ -8,15 +8,15 @@
  */
 public class DataImportController {
-    private final DataImportInputBoundary INPUTBOUNDARY;
+    private final DataImportInputBoundary inputBoundary;
```
capitalize final variables

Bensonwu02 replied:
same as above

Wizard603 commented on src/main/java/controllers/EntryUndoController.java:
```diff
@@ -9,7 +9,7 @@
 public class EntryUndoController {
 
-    private EntryUndoInputBoundary entryUndo;
+    private final EntryUndoInputBoundary entryUndo;
```
capitalize final variables

Wizard603 commented on src/main/java/entities/DataPoint.java:
```diff
@@ -8,9 +8,9 @@
-    private final DateTimeFormatter FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
-    private final double VALUE;
-    private final LocalDateTime DATE;
+    private final DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
```
capitalize final variables

Bensonwu02 replied:
same as above

Wizard603 reviewed (changes requested):
capitalize final variables

Bensonwu02 committed:
Changed exception handling for data import to display more specific message
Updated README to mention jar file

Wizard603 commented on src/main/java/use_cases/DataImporter.java:
```diff
@@ -89,10 +89,10 @@ private ImportResponseModel read() {
             return presenter.prepareFailView("Folder either does not contain csv files or does not have access.");
-        } catch (ParseException e) {
-            return presenter.prepareFailView("Bad data.");
+        } catch (RuntimeException | ParseException e) {
```
Thank you for making the message more clear

Wizard603 reviewed (dismissed):
apparently static finals can be non capitalized

Bensonwu02 committed:
Added fail case for add metric that rejects metric names containing symbols and/or numbers
Updated test cases to reflect these changes
Updated README to reflect these changes

Wizard603 reviewed (approved):
Good catch on the metric names
